.buildkite/commands/unit-tests-ios.sh

```diff
@@ -0,0 +1,7 @@
+#!/bin/bash -eu
+
+echo "--- :npm: Install Node dependencies"
+npm ci --unsafe-perm --prefer-offline --no-audit --no-progress
+
+echo "--- :node: Lint and Unit Tests"
+CHECK_CORRECTNESS=false CHECK_TESTS=true TEST_RN_PLATFORM=ios ./bin/ci-checks-js.sh
```

Comment:
`unit-tests-android.sh` and ``unit-tests-ios.sh` could have been the same script and have switched platform based on an argument. I decided to split them because I imagine they'll get more platform specific behavior in the future.